smart-contracts/wasm-chain-integration/src/v0/mod.rs

```diff
@@ -26,9 +26,9 @@ impl Logs {
     /// - 0 if data was not logged because it would exceed maximum number of
     ///   logs
     /// - 1 if data was logged.
-    pub fn log_event(&mut self, event: Vec<u8>) -> i32 {
+    pub fn log_event(&mut self, event: Vec<u8>, limit_num_logs: bool) -> i32 {
         let cur_len = self.logs.len();
-        if cur_len < constants::MAX_NUM_LOGS {
+        if !limit_num_logs || cur_len < constants::MAX_NUM_LOGS {
```

Comment:
For extra safety please check that `cur_len < u32::MAX` so that we can serialize this thing.